connection_configuration_test.go

```diff
@@ -0,0 +1,226 @@
+package gosnowflake
+
+import (
+	"io/fs"
+	"os"
+	path "path/filepath"
+	"testing"
+	"time"
+)
+
+func TestLoadConnectionConfig_Default(t *testing.T) {
+	err := os.Chmod("./test_data/connections.toml", 0600)
+	assertNilF(t, err, "The error occurred because you cannot change the file permission")
+
+	os.Setenv("SNOWFLAKE_HOME", "./test_data")
+
+	cfg, err := LoadConnectionConfig()
+
+	assertNilF(t, err, "The error should not occur")
+	assertEqualF(t, cfg.Account, "snowdriverswarsaw.us-west-2.aws")
+	assertEqualF(t, cfg.User, "test_user")
+	assertEqualF(t, cfg.Password, "test_pass")
+	assertEqualF(t, cfg.Warehouse, "testw")
+	assertEqualF(t, cfg.Database, "test_db")
+	assertEqualF(t, cfg.Schema, "test_go")
+	assertEqualF(t, cfg.Protocol, "https")
+	assertEqualF(t, cfg.Port, 443)
+}
+
+func TestLoadConnectionConfig_OAuth(t *testing.T) {
+	err := os.Chmod("./test_data/connections.toml", 0600)
+	assertNilF(t, err, "The error occurred because you cannot change the file permission")
+
+	os.Setenv("SNOWFLAKE_HOME", "./test_data")
+	os.Setenv("SNOWFLAKE_DEFAULT_CONNECTION_NAME", "aws-oauth")
+	cfg, err := LoadConnectionConfig()
+
+	assertNilF(t, err, "The error should not occur")
+	assertEqualF(t, cfg.Account, "snowdriverswarsaw.us-west-2.aws")
+	assertEqualF(t, cfg.User, "test_user")
+	assertEqualF(t, cfg.Password, "test_pass")
+	assertEqualF(t, cfg.Warehouse, "testw")
+	assertEqualF(t, cfg.Database, "test_db")
+	assertEqualF(t, cfg.Schema, "test_go")
+	assertEqualF(t, cfg.Protocol, "https")
+	assertEqualF(t, cfg.Authenticator, AuthTypeOAuth)
+	assertEqualF(t, cfg.Token, "token_value")
+	assertEqualF(t, cfg.Port, 443)
+}
+
+func TestLoadConnectionConfigWitNonExisitngDSN(t *testing.T) {
+	err := os.Chmod("./test_data/connections.toml", 0600)
+	assertNilF(t, err, "The error occurred because you cannot change the file permission")
+
+	os.Setenv("SNOWFLAKE_HOME", "./test_data")
+	os.Setenv("SNOWFLAKE_DEFAULT_CONNECTION_NAME", "unavailableDSN")
+
+	_, err = LoadConnectionConfig()
+	assertNotNilF(t, err, "The error should occur")
+
+	driverErr, ok := err.(*SnowflakeError)
+	assertTrueF(t, ok, "This should be a Snowflake Error")
+	assertEqualF(t, driverErr.Number, ErrCodeFailedToFindDSNInToml)
+}
+
+func TestLoadConnectionConfigWithTokenFileNotExist(t *testing.T) {
+	err := os.Chmod("./test_data/connections.toml", 0600)
+	assertNilF(t, err, "The error occurred because you cannot change the file permission")
+
+	os.Setenv("SNOWFLAKE_HOME", "./test_data")
+	os.Setenv("SNOWFLAKE_DEFAULT_CONNECTION_NAME", "aws-oauth-file")
```

Comment:
Based on codecov report, it's missing a test for reading token from token_file_path (only the error cases have been tested). Is it something can be tested?